boxcli/midcobra/debug.go

```diff
@@ -62,5 +63,10 @@ func (d *DebugMiddleware) postRun(cmd *cobra.Command, args []string, runErr erro
 	}
 
 	st := debug.EarliestStackTrace(runErr)
-	debug.Log("Error: %v\n%+v", runErr, st)
+	debug.Log("Error: %v\nexecutionID:%s\n%+v\n", runErr, d.executionID, st)
+}
+
+func (d *DebugMiddleware) withExecutionID(execID string) Middleware {
+	d.executionID = execID
+	return d
 }
```

Comment:
Should this be optional? Why not just always generate an execution ID? If you do keep it optional, there should probably be a check to see if it's empty before printing it in `postRun`.

Reply:
its optional in the sense that I can see some middleware not caring about handling this value (i.e. saving it and using it)

The midcobra framework is responsible for calling withExecutionID, so I don't think individual middlewares need to worry about it not being called. In debug middleware's case, if postRun would has an empty value, then it'll just print as empty-string....which is fine i guess?